pkg/pillar/cmd/upgradeconverter/parseconfig.go

```diff
@@ -98,16 +98,20 @@ func parseAppInstances(config *zconfig.EdgeDevConfig) ([]driveAndVolumeRef, erro
 		davr := make([]driveAndVolumeRef, len(drives))
 		for i, drive := range drives {
 			image := drive.GetImage()
-			imageID, _ := uuid.FromString(image.Uuidandversion.Uuid)
+			if image == nil {
```

Comment:
I think in this case it is sufficient to do a continue here i.e., skip this image. That is because this feature of the controller only appears when there is also a VolumeRef and we only need this conversion support when there is no volumeref.

Can you test such a fix?

Reply:
Good point, I made modification to not include such images.

No, unfortunately I did not tested it. Seems better to fix it on cloud side firstly because it affects existing nodes and may make them non-updatable (without manual deleting of saved config file).